### PR TITLE
Add pointer of ValueBase instead of the Instruction into SymbolTable and NodeMap

### PR DESCRIPTION
**Pull request by bryantam**
For instructions like
```
%5 = apply %4(%3, %2) : $@convention(method) (Builtin.Int2048, @thin Int.Type) -> Int // user: %6 line:2:10
  store %5 to [trivial] %1 : $*Int                // id: %6 line:2:10
```
When the StoreInst try to access %5, it is looking for the ValueBase portion of the ApplyInst (which is 0x48 bytes off *AI). Thus, Instructions should be casted into ValueBase before adding to SymbolTable or NodeMap.

### CODE REVIEW

File: lib/WALASupport/SILWalaInstructionVisitor.cpp [hunk 1]
```diff
@@ -300,7 +300,7 @@ jobject SILWalaInstructionVisitor::visitAllocExistentialBoxInst(AllocExistential
 
     auto name = "ExistentialBox:" + 
       AEBI->getFormalConcreteType().getString() + "->" + AEBI->getExistentialType().getAsString();
-    SymbolTable.insert(((char *)AEBI) + sizeof(SILInstruction) , name);
+    SymbolTable.insert(static_cast<ValueBase *>(AEBI), name);
```
Comment (karimhamdanali) [inline lib/WALASupport/SILWalaInstructionVisitor.cpp:303]: Aren't these completely different values? In the first one, we're also annotating the string with the size of the SILInstruction, while in the second, we're adding the ValueBase for AEBI.

File: lib/WALASupport/SILWalaInstructionVisitor.cpp [hunk 2]
```diff
@@ -455,7 +455,7 @@ jobject SILWalaInstructionVisitor::visitProjectExistentialBoxInst(ProjectExisten
     llvm::outs() << "Operand addr " << PEBI->getOperand().getOpaqueValue() << "\n";
   }
   if (SymbolTable.has(PEBI->getOperand().getOpaqueValue())) {
-    SymbolTable.duplicate(((char *)PEBI) + sizeof(SILInstruction), SymbolTable.get(PEBI->getOperand().getOpaqueValue()).c_str());
+    SymbolTable.duplicate(static_cast<ValueBase *>(PEBI), SymbolTable.get(PEBI->getOperand().getOpaqueValue()).c_str());
```
Comment (karimhamdanali) [inline lib/WALASupport/SILWalaInstructionVisitor.cpp:458]: Same comment as AEBI.

**Review by karimhamdanali — COMMENTED** [review]
How does this change affect the way the values/symbols are consumed by the SymbolTable and NodeMap in WALA?

File: lib/WALASupport/SILWalaInstructionVisitor.cpp (hunk 1 above)
Comment (bryantam) [reply lib/WALASupport/SILWalaInstructionVisitor.cpp:303]: Annotating the string with the size of the SILInstruction was the change I made in my last pull request, which is essentially doing the same thing as the cast. I found it cleaner and easier to understand with a cast than annotating a pointer with some value.

Comment (bryantam) [reply lib/WALASupport/SILWalaInstructionVisitor.cpp:303]: I didn't realize that a cast will do the work when I was making my last PR, and that's why I made the old one. Please consider this as a fix to my last PR for AEBI and PEBI.

Comment (karimhamdanali) [reply lib/WALASupport/SILWalaInstructionVisitor.cpp:303]: Ah, I see your point now.

**Review by karimhamdanali — APPROVED** [review]
Looks good to me now. Bryan, thanks for the clarifications.

**Pull request merged**